plugins/hls-call-hierarchy-plugin/README.md

```diff
@@ -0,0 +1,23 @@
+# Call hierarchy plugin for the [Haskell Language Server](https://github.com/haskell/haskell-language-server#readme)
+
+The call hierarchy plugin can review the code to determine where functions are called and how they relate to other functions.
+
+This plugin is useful when debugging and refactoring code because it allows you to see how different parts of the code are related. And it is more conducive for users to quickly understand their macro architecture in the face of strange code.
+
+## Demo
+
+![Call Hierarchy in Emacs](call-hierarchy-in-emacs.gif)
+
+![Call Hierarchy in VSCode](call-hierarchy-in-vscode.gif)
+
+## Prerequisite
+None. You can experience the whole feature without any setting.
+
+## Configuration
+Enable default. You can disable it in the setting file (JSON format) whenever you like.
```

Comment:
@jneira I am not sure how to point out where the setting is precisely, maybe we can link it to a unified place(lots of plugins need this)

Reply:
mmm you can the entire hls server configuration with 2 commands:
* `haskell-language-server vscode-extension-schema`: it outputs a json usable in the vscode extension package.json config to add them auto to the extension config ui in vscode
* `haskell-language-server generate-default-config`: it outputs the default values of that config

So the configuration for hlint, for example, would be:

```json
{
  "haskell.plugin.hlint.diagnosticsOn": false
}
```
If you put a file with that content in a `settings.json` int the project root dir, you will disable hlint

... but i tried both commands with last version and it does not mention any hls-call-hierarchy plugin.. 🤔 

I think something has to be done to "register" the plugin and include its config in the output, but dont remember right now (i think @berberman did that nice addition to the cli)

Reply:
> but i tried both commands with last version and it does not mention any hls-call-hierarchy plugin

Ah, that's because
https://github.com/haskell/haskell-language-server/blob/047ffc366461470a27ea47c2042c76bbf97e1d15/hls-plugin-api/src/Ide/Plugin/ConfigUtils.hs#L87-L95

and

https://github.com/haskell/haskell-language-server/blob/047ffc366461470a27ea47c2042c76bbf97e1d15/hls-plugin-api/src/Ide/Plugin/ConfigUtils.hs#L117-L124

need update for new call hierarchy handlers

Reply:
@July541 could you have the chance of update those definitions, to complete the hierarchy plugin integration?

Reply:
Sure, I'll do it.